src/renderers/dom/client/wrappers/ReactDOMSelect.js

```diff
@@ -58,6 +58,28 @@ function warnIfValueIsNull(props) {
   }
 }
 
+function warnIfDuplicateValues(inst) {
```

Comment:
You would create a new file for this warning. #7040 is just a reference for what it might look like. You can maybe name it something like `ReactDOMDuplicateSelectValuesDevtool.js` to be consistent with the current naming scheme.